NAVIGATION_DIAGRAM.md

```diff
@@ -0,0 +1,212 @@
+# Controller Navigation Flow Diagram
+
+## Visual Layout
+
+```
+┌─────────────────────────────────────────────────────────────┐
+│                     CRAFTING UI                              │
+├─────────────────────────────────────────────────────────────┤
+│                                                               │
+│  COLLECTIBLES AREA (top)                                     │
+│  ┌──────┐  ┌──────┐  ┌──────┐  ┌──────┐  ┌──────┐          │
+│  │ [C1] │  │  C2  │  │  C3  │  │  C4  │  │  C5  │  ...     │
+│  └──────┘  └──────┘  └──────┘  └──────┘  └──────┘          │
+│    ▲───────►────────►────────►────────►────────►            │
+│    ◄───────◄────────◄────────◄────────◄────────◄            │
+│      LEFT/RIGHT navigation within collectibles               │
+│                                                               │
+│                        ▼ DOWN                                 │
+│                        ▲ UP                                   │
+│                                                               │
+│  SLOTS/BUTTONS AREA (bottom)                                 │
+│  ┌──────┐  ┌──────┐  ┌──────┐    ┌──────┐                  │
+│  │[Slot1]│  │ Slot2│  │ Slot3│    │Output│                  │
+│  └──────┘  └──────┘  └──────┘    └──────┘                  │
+│    ▲───────►────────►                                        │
+│    ◄───────◄────────◄                                        │
+│      LEFT/RIGHT navigation within slots                      │
```

Comment:
Right now you can only remove the last slot so we shouldn't support navigating left or right on the slots. We should automatically select the last slot.